src/Microsoft.PowerShell.Commands.Utility/commands/utility/FormatAndOutput/format-hex/Format-Hex.cs

```diff
@@ -423,7 +432,11 @@ private byte[] ConvertToBytes(object inputObject)
                 int index = 0;
                 foreach (dynamic obj in (Array)inputObject)
                 {
-                    if (elementSize == 1)
+                    if (isBool)
+                    {
+                        result[index] = (byte)(obj ? 1 : 0);
```

Comment:
Can you share a screenshot to demo the change?

Reply:
Sure!

### Before

![image](https://user-images.githubusercontent.com/32407840/72406995-1e9e4900-372c-11ea-9101-3f42ef6fa853.png)

(ignore the bugged label on the first block here; the values in it are correct here)

### After

![image](https://user-images.githubusercontent.com/32407840/72407004-2958de00-372c-11ea-8fde-fe9d63443c4e.png)

Reply:
I think it misleads. Because the cmdlet does a dump I'd prefer to see _real_ size.

Reply:
I have no qualms about reverting it, I just see no value in including padding that is apparently only added as a performance optimization.

In practice booleans are generally treated as single-bit values; I think this representation is the most useful here to us . In the 1st screenshot above, it's significantly harder to determine where one value begins and the other ends in the first block there. It's much easier to see what's going on in the 2nd screenshot in my opinion. 🙂 

I'm happy to defer this decision if need be, the priority here _is_ the bug fix; I can pull out the commit if that's what we need to do here. Just trying to make sure it's as user friendly as possible. 😊

Reply:
Also, for what it's worth:

https://sharplab.io/#v2:C4LgTgrgdgNAJiA1AHwAICYCMBYAUKgBgAJVMA6AJWmAEsBbAUzIEkpgGwB7ABwGUOAbjQDGDAM4BuPKgDMJdEQDCRAN54iG+QrW5NekpgCcACgBEvGgC8GRTgDMiAI06cANiFUEAvqZhExVgz2xs5uAJRhUrr6GqQmpgCyAIZgYgAWSa6uDHD+gbYOoe6ePn7JqRmuZBbWAPJ2xsAAntxBDUURkeqaXnheQA===

That snippet illustrates the difference; for the intents of anyone _using_ the code, a bool is a single byte... the padding difference really only has an impact if you're doing memory analysis of some form or other. At least that's what I gather from a bit of searching on the topic. 🙂

Reply:
Further reading, since I was curious:

https://stackoverflow.com/a/28515361

(especially relevant paragraph) 

> The _actual_ size of a C# `bool` is one byte. A strong design-goal of the CLR is that you cannot find out. Layout is an implementation detail that depends on the processor too much. Processors are very picky about variable types and alignment, wrong choices can significantly affect performance and cause runtime errors. By making the layout undiscoverable, .NET can provide a universal type system that does not depend on the actual runtime implementation.

Reply:
I consider fxh as a memory dumper and in memory the bool is 4 byte.
It is better to discuss this in new issue or PR.

Reply:
Definitely fair. 🙂 

I'll revert this particular change for now ^^